### PR TITLE
I116 shared quiz models app

### LINKED ISSUE #116: Shared quiz models app
## Basic Information
There are several quiz-related backend issues which are each separate (and should be), however rely on some shared underlying models, particularly to do with quizzes themselves. To connect all of these apps together and prevent code duplication, a new `shared_models` app will be made to store these shared quiz models, and other apps will refer to these models rather than having their own local copies.

## Other Information
* Essentially a merge of the models used in #72, #73 and #74

### PR DESCRIPTION
**Pull request by cjnbennett**
## Change Summary
Implemented several quiz and statistics models (and their respective serializers) that will be shared between the various quiz apps (#72, #73, #74). This issue is a priority so that the other quiz-related issues can continue work with a definite set of models. The documentation goes into the details of these models and how they are intended to be used.

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [x] The pull request title has an issue number
- [x] The change works by "Smoke testing" or quick testing
- [x] The change has tests
- [x] The change has documentation

## Other Information
**[Is there anything in particular in the review that I should be aware of?]**

# Related issue

- Resolve #116

### CODE REVIEW

**Comment by cjnbennett** [comment]
There are still some linting "issues" in the code (hence why the linting test failed), those are files automatically generated by Django for the purpose of database migrations, so I have not touched them. There are no linting issues in any other files.

**Comment by jsun1590** [comment]
> There are still some linting "issues" in the code (hence why the linting test failed), those are files automatically generated by Django for the purpose of database migrations, so I have not touched them. There are no linting issues in any other files.

@cjnbennett Could you still run linting on those in the future please?

You can do `black . --line-length 79 --preview`

File: server/api/apps/shared_models/apps.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from django.apps import AppConfig
```
Comment (jsun1590) [inline server/api/apps/shared_models/apps.py:1]: apps.py is not needed if you are directly installing the app in settings.py and not changing other configurations. Alternatively, when installing the app reference the config file.
https://stackoverflow.com/a/46346843

File: server/api/apps/shared_models/models/quiz_models.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+from django.db import models
+from django.utils.translation import gettext_lazy as _
```
Comment (jsun1590) [inline server/api/apps/shared_models/models/quiz_models.py:2]: Translations are not included in the MVP scope currently, so we don't really need to use this

File: server/api/apps/shared_models/models/quiz_models.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+from . import defines
+from django.contrib.auth.models import User
```
Comment (jsun1590) [inline server/api/apps/shared_models/models/quiz_models.py:4]: We are using a custom User model in the project. Please see this: https://docs.djangoproject.com/en/4.0/topics/auth/customizing/#referencing-the-user-model

File: server/api/apps/shared_models/models/statistics_models.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+from django.db import models
+from .quiz_models import Question, Answer
+from django.contrib.auth.models import User
```
Comment (jsun1590) [inline server/api/apps/shared_models/models/statistics_models.py:3]: https://docs.djangoproject.com/en/4.0/topics/auth/customizing/#referencing-the-user-model

**Review by jsun1590 — CHANGES_REQUESTED** [review]
Well done, great job overall! Just a couple of small changes needed :)

**Review by jsun1590 — APPROVED** [review]
Thanks, looks great! Feel free to merge :+1:

**Pull request merged**